src/wallet/wallet2.cpp

```diff
@@ -877,7 +877,7 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
       LOG_PRINT_L0("Public key wasn't found in the transaction extra. Skipping transaction " << txid);
       if(0 != m_callback)
 	m_callback->on_skip_transaction(height, txid, tx);
-      return;
+      break;
```

Comment:
So, when does this happen? There is no pubkey, but the user could have spent money in the transaction?

This function could use a refactor in smaller sub-functions for some sanity ...

Reply:
I did, some old bug I don't remember about now :)
But it was causing my wallet to have some available outputs which were in fact spent.